tests/ga/test_monitor.py

```diff
@@ -342,18 +344,19 @@ def test_heartbeats(self,
         self.assertEqual(0, patch_send_telemetry_metrics.call_count)
         self.assertEqual(0, patch_poll_telemetry_metrics.call_count)
 
-        monitor_handler.start()
-        time.sleep(1)
-        self.assertTrue(monitor_handler.is_alive())
+        with patch.object(monitor_handler, 'protocol'):
```

Comment:
What is this patching? Are you forcefully returning a mock when looking at self.protocol?

Reply:
Or you intend to test it with protocol to be None

Reply:
Yes, I'm forcefully returning a mock for the monitor_thread.protocol object. This was not required before my changes because we weren't calling the `protocol.update_goal_state()` before my changes. This change in behaviour was breaking the test due to which I had to mock it